packages/plugins/scripts-runner/src/index.ts

```diff
@@ -0,0 +1,171 @@
+    const forceTracking = options.forceTracking;
+    const scriptName = path.basename(options.target);
+    const scriptDirectory = path.basename(path.dirname(options.target));
+    const scriptTracked = await this.embark.events.request2('deployment-tracker:script:tracked', { scriptName });
+
+    if (scriptTracked && (scriptDirectory === this.embark.config.embarkConfig.migrations || forceTracking)) {
+      this.embark.logger.info(`.. ${scriptName} already done`);
+      return callback();
+    }
+
+    this.embark.logger.info(`.. ${scriptName} running....`);
+    const scriptToRun = require(options.target);
+
+    try {
+      const result = await scriptToRun(options.dependencies);
+      this.embark.logger.info(`.. finished.`);
+      this.plugins.emitAndRunActionsForEvent(ScriptsRunnerEvent.Executed, {
+        scriptName,
+        scriptDirectory,
+        forceTracking,
+        migrationsDirectory: this.embark.config.embarkConfig.migrations
+      }, err => {
+        if (err) {
+          return callback(err);
+        }
+        callback(null, result);
+      });
+    } catch (e) {
+      const error = e instanceof Error ? e : new Error(e);
+      callback(new ScriptExecutionError(options.target, error));
+    }
+  }
+
+  private async executeAll(options: ExecuteOptions, callback: Callback<any>) {
+    const target = options.target;
+    const files = await fsPromises.readdir(target);
+    const results: any[] = [];
+
+    eachSeries(files, (file, cb) => {
+      const targetPath = !path.isAbsolute(target) ? dappPath(target, file) : path.join(target, file);
+
+      stat(targetPath, (err, fstat) => {
+        if (err) {
+          return cb(err);
+        }
+        if (fstat.isFile()) {
+          return this.executeSingle({
+            target: targetPath,
+            dependencies: options.dependencies,
+            forceTracking: options.forceTracking
+          }, (e, result) => {
+            if (e) {
+              return cb(e);
+            }
+            results.push(result);
+            cb();
+          });
+        }
+        cb();
+      });
+    }, (err) => {
+      if (err) {
+        return callback(err);
+      }
+      callback(null, results);
+    });
+  }
+
+  private async getScriptDependencies() {
+    const contracts = await this.embark.events.request2('contracts:list');
+
+    const dependencies: ScriptDependencies  = {
+      logger:  this.embark.logger,
+      web3: null,
+      contracts: {}
+    };
+
+    dependencies.web3 = await this.web3;
+
+    for (const contract of contracts) {
+      const registeredInVM = this.checkContractRegisteredInVM(contract);
+      if (!registeredInVM) {
+        await this.embark.events.request2("embarkjs:contract:runInVm", contract);
+      }
+      const contractInstance = await this.embark.events.request2("runcode:eval", contract.className);
+      dependencies.contracts[contract.className] = contractInstance;
```

Comment:
I'm not sure I understand how running `ContractName` in the VM gets the dependencies? Seems like it just gets the instance

Reply:
NVM, I just understood by reading the docs. You get the dependencies that get passed to the migration script function. Don't we already do that in the `hooks` functions? Is there a possibility of using the same mechanism?

Reply:
Yes, there is and I've also left a comment there that this could be fetched from another plugin.

However, now that @iurimatias mentioned plugins shouldn't depend on other plugins, I guess it's actually desired to have this logic here.